src/openrct2/ride/Ride.cpp

```diff
@@ -7502,19 +7501,20 @@ static void ride_update_vehicle_colours(int32_t rideIndex)
             {
                 case RIDE_COLOUR_SCHEME_ALL_SAME:
                     colours = ride->vehicle_colours[0];
-                    coloursExtended = ride->vehicle_colours_extended[0];
+                    coloursExtended = ride->vehicle_colours[0].Ternary;
                     break;
                 case RIDE_COLOUR_SCHEME_DIFFERENT_PER_TRAIN:
                     colours = ride->vehicle_colours[i];
-                    coloursExtended = ride->vehicle_colours_extended[i];
+                    coloursExtended = ride->vehicle_colours[i].Ternary;
                     break;
                 case RIDE_COLOUR_SCHEME_DIFFERENT_PER_CAR:
                     colours = ride->vehicle_colours[std::min(carIndex, MAX_CARS_PER_TRAIN - 1)];
-                    coloursExtended = ride->vehicle_colours_extended[std::min(carIndex, MAX_CARS_PER_TRAIN - 1)];
+                    coloursExtended = ride->vehicle_colours[std::min(carIndex, MAX_CARS_PER_TRAIN - 1)].Ternary;
                     break;
             }
 
-            vehicle->colours = colours;
+            vehicle->colours.body_colour = colours.Body;
+            vehicle->colours.trim_colour = colours.Trim;
             vehicle->colours_extended = coloursExtended;
```

Comment:
This variable can also be set from `colours`. By doing this, the `coloursExtended` can be removed.